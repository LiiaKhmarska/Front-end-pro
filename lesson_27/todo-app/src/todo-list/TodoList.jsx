import { useState } from 'react';
import { TodoForm } from './TodoForm';
import { TodoItem } from './TodoItem';

export function TodoList() {
    const [todos, setTodos] = useState([]);

    function addTask(taskInput) {        
        if (taskInput) {
            const newTask = {
                id: Date.now(),
                task: taskInput,
                completed: false
            };
        setTodos([...todos, newTask])
        }   
    }

    function removeTask(id) {
        setTodos([...todos.filter((todo) => todo.id!==id)])
    }

     function toggleTask(id) {
        setTodos([...todos.map(todo => todo.id===id ?{...todo, completed:!todo.completed}:{...todo})])
    }

    return (
        <div className="todo-container">
            <h1>To-Do List</h1>
            <TodoForm addTask={addTask}/>
            {todos.map((todo) => (
                <TodoItem todo={todo} key={todo.id}
                    removeTask={removeTask}
                    toggleTask={toggleTask} />
            ))}
        </div>
    )
}