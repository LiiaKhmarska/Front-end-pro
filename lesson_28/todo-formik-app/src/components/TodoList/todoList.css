.todo-container {
    width: 80%;
    min-height: 70vh;
    /* background-color: rgba(127, 255, 251, 0.195); */
    border-radius: 3%;
    margin: 10px auto;
    padding: 2%;
    border: 2px solid lightgrey;
    box-shadow: 5px 5px 11px 3px gray;
}
h1 {
    text-align: center;
}