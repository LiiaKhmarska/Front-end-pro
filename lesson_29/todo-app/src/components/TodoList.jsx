import './todo.css';
import { TodoForm } from './TodoForm';
import { TodoItem } from './TodoItem';
import { TodoCounter } from './TodoCounter';
import { useSelector } from 'react-redux';

export function TodoList() {

    const todos = useSelector((state) => state.todos.todos)
    

       return (
        <div className="container">
            <h1>TODO</h1>
            <TodoForm />
            <h2>TODOS</h2>
            <hr />
            <div className='todos-list-container'>
            
            {todos.map((todo) => (
                <TodoItem todo={todo} key={ todo.id} />
            ))}
                
            </div>
            <TodoCounter counter ={todos.length} />
        </div>
    )
}