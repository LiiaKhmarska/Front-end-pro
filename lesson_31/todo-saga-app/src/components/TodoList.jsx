import './todo.css';
import { TodoForm } from './TodoForm';
import { TodoItem } from './TodoItem';
import { TodoCounter } from './TodoCounter';
import { useDispatch, useSelector } from 'react-redux';
import { selectTodos, selectStatus } from '../ducks/todos/todos.slice';
import { useEffect } from 'react';
import { FETCH_TODOS } from '../ducks/todos/todos.actions';

export function TodoList() {
    const dispatch = useDispatch();
    const todos = useSelector(selectTodos);
    
    
    useEffect(() => {
        dispatch(FETCH_TODOS())     
    },[])

       return (
        <div className="container">
            <h1>TODO</h1>
            <TodoForm />
            <h2>TODOS</h2>
            <hr />
            <div className='todos-list-container'>
            
            {todos.map((todo) => (
                <TodoItem todo={todo} key={ todo.id} />
            ))}
                
            </div>
            <TodoCounter counter = {todos.length} />
        </div>
    )
}