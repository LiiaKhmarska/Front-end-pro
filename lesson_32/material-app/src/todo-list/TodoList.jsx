
import { TodoForm } from './TodoForm';
import { TodoItem } from './TodoItem';
import { TodoCounter } from './TodoCounter';
import { useSelector } from 'react-redux';
import { Header } from '@components/Header'
import {List, Typography} from '@mui/material'


export function TodoList() {

    const todos = useSelector((state) => state.todos.todos)
    

    return (
           <>
           <Header />
          
        <div className="container">
            <Typography variant="h3" gutterBottom sx={{ mt: 5 }}>
            TODO
      </Typography>
            <TodoForm />
            <Typography variant="h4" sx={{ mt: 2 }}> All your todos</Typography>
            
            <List dense={true}>
            
            {todos.map((todo) => (
                <TodoItem todo={todo} key={ todo.id} />
            ))}
                
            </List>
            <TodoCounter counter ={todos.length} />
            </div>
             </>
    )
}