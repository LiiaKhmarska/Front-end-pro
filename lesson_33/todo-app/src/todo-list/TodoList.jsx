
import { TodoForm } from './TodoForm';
import { TodoItem } from './TodoItem';
import { TodoCounter } from './TodoCounter';
import { useSelector } from 'react-redux';
import { Box, List, Typography } from '@mui/material'
import { selectTodos } from '../store/todoSlice';


export function TodoList() {
    const todos = useSelector(selectTodos)
    
    return (
           <>
        <Box  sx={{width:'80%', mx:'10%'}}>
            <Typography variant="h3" gutterBottom sx={{ mt: 5 }}>
            TODO
      </Typography>
            <TodoForm />
            <Typography variant="h4" sx={{ mt: 2 }}> All your todos</Typography>
            
            <List dense={true} data-testid='todo-list'>
            
            {todos.map((todo) => (
                <TodoItem todo={todo} key={ todo.id} />
            ))}
                
            </List>
            <TodoCounter counter ={todos.length} />
            </Box>
             </>
    )
}