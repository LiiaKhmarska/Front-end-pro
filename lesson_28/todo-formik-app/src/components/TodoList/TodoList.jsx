import './todoList.css';
import { CustomForm } from '../Form/Form';
import { useState } from 'react';
import { TaskItem } from '../TaskItem/TaskItem';

export function TodoList() {
   
    const [tasks, setTasks] = useState([]);

    function addTask(inputData) {
          
        if (inputData) {
            const newTask = {
                id: Date.now(),
                title:inputData.title,
                task: inputData.task,
                completed: false
            };
        setTasks([...tasks, newTask])
        }   
    }
    
    function removeTask(id) {       
        setTasks([...tasks.filter((task) => task.id!==id)])
    }

     function toggleTask(id) {
        setTasks([...tasks.map(task => task.id===id ?{...task, completed:!task.completed}:{...task})])
    }


    return (
        <div className="todo-container">
            <h1>To-Do List</h1>
            <CustomForm addTask={addTask} />
            {
                tasks.map((task) => (
                
                    <TaskItem task={task} key={ task.id }
                        toggleTask={toggleTask}
                        removeTask={removeTask}
                    />
                ))
           }

           
        </div>
    )
}