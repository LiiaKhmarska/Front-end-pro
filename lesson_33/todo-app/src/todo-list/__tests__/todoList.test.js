import { screen, render, toMatchSnapshot } from "@testing-library/react";
import { describe, expect, jest } from "@jest/globals";
import { TodoList } from "../TodoList";
import { useSelector } from "react-redux";

jest.mock("react-redux");

const mockedInitialState = {
  todos: [{ id: 2147, content: "Make todo-list", checked: true }],
};

describe("todo list", () => {
  it("Should create todo list with defult todo ", () => {
    useSelector.mockReturnValue(mockedInitialState.todos);
    const component = render(<TodoList />);
    expect(component).toMatchSnapshot();
  });

  it("Should have heading TODO ", () => {
    render(<TodoList />);
    const h3 = screen.getByRole("heading", { level: 3 });
    expect(h3).toBeInTheDocument();
    expect(h3).toHaveTextContent(/todo/i);
  });

  it("Should have heading 'All your todos' ", () => {
    render(<TodoList />);
    const h4 = screen.getByRole("heading", { level: 4 });
    expect(h4).toBeInTheDocument();
    expect(h4).toHaveTextContent(/todo/i);
  });

  it("should have list of todos", () => {
    render(<TodoList />);
    const listItem = screen.queryByTestId("todo-list");
    expect(listItem).toBeInTheDocument();
    expect(listItem).toHaveTextContent(/todo/i);
  });

  it('Should have "Total" for count todos', () => {
    render(<TodoList />);
    const total = screen.queryByTestId("todo-counter");
    expect(total).toBeInTheDocument();
    expect(total).toHaveTextContent("Total:");
  });
});
