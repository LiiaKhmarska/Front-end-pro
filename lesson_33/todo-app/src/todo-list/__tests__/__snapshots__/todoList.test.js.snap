// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`todo list Should create todo list with defult todo  1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-1y7pth8"
      >
        <h3
          class="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom css-142ad5c-MuiTypography-root"
        >
          TODO
        </h3>
        <form
          autocomplete="off"
          class="MuiBox-root css-sl4z9f"
          data-testid="add-todo-form"
          name="form"
          novalidate=""
        >
          <div
            class="MuiFormControl-root MuiTextField-root css-1xp5r68-MuiFormControl-root-MuiTextField-root"
            style="flex-grow: 1;"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-19qnlrw-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="outlined-basic"
              id="outlined-basic-label"
            >
              New task
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-1ljffdk-MuiFormLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-quhxjy-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
                id="outlined-basic"
                name="task"
                placeholder="Add task"
                required=""
                type="text"
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-18p5xg2-MuiNotchedOutlined-root-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1n64csd-MuiNotchedOutlined-root"
                >
                  <span>
                    New task
                     
                    *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <button
            aria-label="Add task"
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-cz8tbf-MuiButtonBase-root-MuiButton-root"
            data-mui-internal-clone-element="true"
            data-testid="btn-add"
            tabindex="0"
            type="submit"
          >
            Add Task
          </button>
        </form>
        <h4
          class="MuiTypography-root MuiTypography-h4 css-u78otf-MuiTypography-root"
        >
           All your todos
        </h4>
        <ul
          class="MuiList-root MuiList-padding MuiList-dense css-b7dj7n-MuiList-root"
          data-testid="todo-list"
        >
          <li
            class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-padding css-2rg357-MuiListItem-root"
          >
            <span
              class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium Mui-checked MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-vuhw6e-MuiButtonBase-root-MuiSwitchBase-root-MuiCheckbox-root"
            >
              <input
                checked=""
                class="PrivateSwitchBase-input css-12xagqm-MuiSwitchBase-root"
                data-indeterminate="false"
                type="checkbox"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                data-testid="CheckBoxIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                />
              </svg>
            </span>
            <div
              class="MuiListItemText-root MuiListItemText-dense css-cfq8qh-MuiListItemText-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-rmuk17-MuiTypography-root"
              >
                Make todo-list
              </p>
            </div>
            <button
              aria-label="Delete"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1ysp02-MuiButtonBase-root-MuiIconButton-root"
              data-mui-internal-clone-element="true"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                data-testid="DeleteIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6zM19 4h-3.5l-1-1h-5l-1 1H5v2h14z"
                />
              </svg>
            </button>
          </li>
        </ul>
        <div
          class="MuiBox-root css-13brihr"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-9fxu5-MuiTypography-root"
            data-testid="todo-counter"
          >
            Total: 
            1
             
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-1y7pth8"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom css-142ad5c-MuiTypography-root"
      >
        TODO
      </h3>
      <form
        autocomplete="off"
        class="MuiBox-root css-sl4z9f"
        data-testid="add-todo-form"
        name="form"
        novalidate=""
      >
        <div
          class="MuiFormControl-root MuiTextField-root css-1xp5r68-MuiFormControl-root-MuiTextField-root"
          style="flex-grow: 1;"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-19qnlrw-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="outlined-basic"
            id="outlined-basic-label"
          >
            New task
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-1ljffdk-MuiFormLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-quhxjy-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
              id="outlined-basic"
              name="task"
              placeholder="Add task"
              required=""
              type="text"
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-18p5xg2-MuiNotchedOutlined-root-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-1n64csd-MuiNotchedOutlined-root"
              >
                <span>
                  New task
                   
                  *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          aria-label="Add task"
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-cz8tbf-MuiButtonBase-root-MuiButton-root"
          data-mui-internal-clone-element="true"
          data-testid="btn-add"
          tabindex="0"
          type="submit"
        >
          Add Task
        </button>
      </form>
      <h4
        class="MuiTypography-root MuiTypography-h4 css-u78otf-MuiTypography-root"
      >
         All your todos
      </h4>
      <ul
        class="MuiList-root MuiList-padding MuiList-dense css-b7dj7n-MuiList-root"
        data-testid="todo-list"
      >
        <li
          class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-padding css-2rg357-MuiListItem-root"
        >
          <span
            class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium Mui-checked MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-vuhw6e-MuiButtonBase-root-MuiSwitchBase-root-MuiCheckbox-root"
          >
            <input
              checked=""
              class="PrivateSwitchBase-input css-12xagqm-MuiSwitchBase-root"
              data-indeterminate="false"
              type="checkbox"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
              data-testid="CheckBoxIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
              />
            </svg>
          </span>
          <div
            class="MuiListItemText-root MuiListItemText-dense css-cfq8qh-MuiListItemText-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-rmuk17-MuiTypography-root"
            >
              Make todo-list
            </p>
          </div>
          <button
            aria-label="Delete"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1ysp02-MuiButtonBase-root-MuiIconButton-root"
            data-mui-internal-clone-element="true"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
              data-testid="DeleteIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6zM19 4h-3.5l-1-1h-5l-1 1H5v2h14z"
              />
            </svg>
          </button>
        </li>
      </ul>
      <div
        class="MuiBox-root css-13brihr"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-9fxu5-MuiTypography-root"
          data-testid="todo-counter"
        >
          Total: 
          1
           
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
